### PR TITLE
IDEM-392

### PR DESCRIPTION
**Pull request by JasminFarhi**
#392 
IMG_7852
IMG_7854

IMG_7851 (2)

### CODE REVIEW

File: app/src/components/PgpFields.tsx [hunk 1]
```diff
@@ -0,0 +1,301 @@
...
+  if (!isCorrectFileType) {
+    throw new Error("Invalid file type : expecting .asc or .key");
+  }
+  const fileContent = await FileSystem.readAsStringAsync(res.uri);
+  return fileContent;
+};
+
+const PgpFields: React.FC<Props> = (props) => {
+  // for user input
+  const [keyText] = React.useState<string>();
```
Comment (pondpiu) [inline app/src/components/PgpFields.tsx:47]: state without setState

File: app/src/components/PgpFields.tsx [hunk 2]
```diff
@@ -0,0 +1,301 @@
...
+                    "Generate new PGP Key and publish",
+                    "cancel"
+                  ],
+                  cancelButtonIndex: 3
+                },
+                async (buttonIndex) => {
+                  if (buttonIndex === 0) {
+                    importMyPrivateKeyFromTextInput(
+                      keyText as string,
+                      props.emailInput as string
```
Comment (pondpiu) [inline app/src/components/PgpFields.tsx:211]: no method to input keyText

File: app/src/hooks/usePpg.ts [hunk 3]
```diff
@@ -19,6 +20,7 @@ type Hooks = {
     email: string
   ) => Promise<void>;
   verifyPublicKey: (email: string | undefined) => Promise<void>;
+  resendVerificationEmail: (email: string) => Promise<void>;
```
Comment (xiangzhe0812) [inline app/src/hooks/usePpg.ts:23]:
```suggestion
  resendVerificationEmail: (email: string) => Promise<boolean>;
```

File: app/src/components/PgpFields.tsx [hunk 4]
```diff
@@ -0,0 +1,301 @@
...
+        if (!content) return;
+        await extractAndLoadKeyPairFromContent(content, email);
+      } catch (error: any) {
+        Alert.alert(
+          AlertTitle.Error,
+          `Failed to extract the Private Key from file \n> ${
+            error?.message ?? "unknown error"
+          }`
+        );
+        console.error(error);
```
Comment (xiangzhe0812) [inline app/src/components/PgpFields.tsx:118]: remove

File: app/package.json [hunk 5]
```diff
@@ -50,7 +50,9 @@
     "react-native-fast-openpgp": "^2.4.2",
     "react-native-gesture-handler": "~2.2.1",
     "react-native-get-random-values": "~1.8.0",
+    "react-native-qrcode-svg": "^6.1.2",
     "react-native-safe-area-context": "4.2.4",
+    "react-native-svg": "12.3.0",
```
[the remark this answers is not in the document]
Comment (pondpiu) [reply app/package.json:55]: Yes, according to doc, `react-native-qrcode-svg` need `react-native-svg` to be installed

File: app/src/components/PgpFields.tsx [hunk 6]
```diff
@@ -0,0 +1,301 @@
...
+  const isCorrectFileType =
+    res.name.endsWith(".asc") || res.name.endsWith(".key");
+  if (!isCorrectFileType) {
+    throw new Error("Invalid file type : expecting .asc or .key");
+  }
+  const fileContent = await FileSystem.readAsStringAsync(res.uri);
+  return fileContent;
+};
+
+const PgpFields: React.FC<Props> = (props) => {
```
[the remark this answers is not in the document]
Comment (pondpiu) [reply app/src/components/PgpFields.tsx]: not a screen, i changed to PgpSection.

File: app/src/hooks/usePpg.ts (hunk 3 above)
Comment (pondpiu) [reply app/src/hooks/usePpg.ts]: no return from function just a alert box, do you mean API endpoint return boolean?

File: app/apis/Api.ts [hunk 7]
```diff
@@ -60,6 +60,10 @@ export default class Api extends HTTPClient {
         }
       }
     );
+
+  public resendVerificationEmail = async (body: { hashedEmail: string }) =>
+    this.post(`user/resend-email`, body);
```
Comment (brettvniekerk) [inline app/apis/Api.ts:65]: needs return type no?

**Review by xiangzhe0812 — APPROVED** [review]

File: app/src/components/PgpSection.tsx [hunk 8]
```diff
@@ -0,0 +1,251 @@
...
+  );
+  const generateAndPublishNewPgpKey = React.useCallback(
+    async (name: string, email: string) => {
+      await generateKeyPair(name, email);
+      await addClaim(ClaimTypeConstants.EmailCredential, { email }, [], false);
+      await loadKeyFromLocalStorage();
+    },
+    [generateKeyPair]
+  );
+  const toggleSwitch = () => setIsActive((previousState) => !previousState);
```
Comment (brettvniekerk) [inline app/src/components/PgpSection.tsx:89]: can this not be `setIsActive(!isActive)`?

File: app/src/hooks/usePpg.ts [hunk 9]
```diff
@@ -100,36 +105,80 @@ const usePgp = (): Hooks => {
...
-            { email: email },
-            [],
-            true
+  const resendVerificationEmail = async (email: string) => {
+    const formattedEmail = email.trim().toLowerCase();
+    await api
+      .resendVerificationEmail({
+        hashedEmail: ethers.utils.hashMessage(formattedEmail)
+      })
+      .then((response) => {
```
Comment (brettvniekerk) [inline app/src/hooks/usePpg.ts:114]: could be `then((response) => response && Alert.alert(etc..))`

Comment (pondpiu) [reply app/src/hooks/usePpg.ts:114]: i think it's more clear this way, though i remove `return` cuz its sounds like we're returning some result

File: app/src/components/PgpSection.tsx (hunk 8 above)
Comment (pondpiu) [reply app/src/components/PgpSection.tsx:89]: true, but i think its fine for now

**Review by pondpiu — APPROVED** [review]

**Review by xiangzhe0812 — APPROVED** [review]

**Pull request merged**